torchvision/models/__init__.py

```diff
@@ -15,4 +15,8 @@
 from .swin_transformer import *
 from .maxvit import *
 from . import detection, optical_flow, quantization, segmentation, video
-from ._api import get_model, get_model_builder, get_model_weights, get_weight, list_models
+
+# We're making Weights and WeightsEnum public for packages like torchgeo who are
+# interested in using them https://github.com/pytorch/vision/issues/7094
+# TODO: we could / should document them publicly as well?
```

Comment:
Yes. Maybe even in this PR?

Reply:
Following up on my comment above about the fact that these are the first developer-facing tools, do you have thoughts on where / how we should document them?
I typically think it shouldn't be part of the already quite dense `models` doc page.

Reply:
Yeah, you are right. With our new maintainer guide, downstream libraries are safe in the sense that we put them under our BC policy. My only concern is discoverability. If we leave it as is, people that want something like this maybe don't know that it is there? Or do we expect them to look into our implementation anyway? cc @adamjstewart

Reply:
Since it's a first, I tempted to leave them undocumented for now and see how it goes. Developers who need those would check the code IMO (or not, rightfully assuming that they're public since they can be imported from `torchvision.models`).

Reply:
From a TorchGeo perspective, we're fine with having to read the source code to figure out how to use the weights. But our docs try to link to the base class docs, so if the base class isn't documented we have to add an ignore. So long term, it would be cool if it was documented, but not a hard requirement.